src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java

```diff
@@ -283,6 +284,7 @@ private static FunctionDescriptor stripNames(FunctionDescriptor function) {
             ValueLayout.JAVA_FLOAT,
             ValueLayout.JAVA_LONG,
             ValueLayout.JAVA_DOUBLE,
-            ValueLayout.ADDRESS
+            ValueLayout.ADDRESS,
+            ValueLayout.JAVA_DOUBLE.withByteAlignment(4) // used by AIX
```

Comment:
We can't add the layout here. The supported layout check essentially tries to verify that the layout used for e.g. `double`, matches the layout of the `C` double on that platform. So, double with 4 byte alignment should not be allowed on other platforms.

I think we should add a linker-specific supported layouts check instead.

I suggest making `checkSupported` non-static, and then adding an abstract method to check if a layout is supported that is called from `checkSupported`. See for instance how it's done for the `cannonicalLayouts` in the panama-foreign repo: https://github.com/openjdk/panama-foreign/blob/foreign-memaccess%2Babi/src/java.base/share/classes/jdk/internal/foreign/abi/x64/windows/Windowsx64Linker.java#L44-L45 this uses a shared factory method in SharedUtils to create the Map, where each linker impl can specify some platform specific layouts, and have it's own instance of the Map. Something similar should work in this case I think.

Reply:
Thanks for the quick response! We'll try (maybe after getting more tests working, there could be more).